packages/web3-types/src/eip6963_types.ts

```diff
@@ -0,0 +1,38 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import type { Web3APISpec } from './web3_api_types.js';
+import type { EIP1193Provider } from './web3_base_provider.js';
+
+export interface EIP6963ProviderInfo {
+	uuid: string;
+	name: string;
+	icon: string;
+	rdns: string;
+}
+
+export interface EIP6963ProviderDetail<API = Web3APISpec> {
+	info: EIP6963ProviderInfo;
+	provider: EIP1193Provider<API>;
+}
+
+export type EIP6963ProviderResponse = Map<string, EIP6963ProviderDetail>;
+
+export interface EIP6963ProvidersMapUpdateEvent extends CustomEvent {
```

Comment:
why these types are moved from web3 to types package?

Reply:
Because I think they should be exported and it seemed like the best way to go about it. I can modify the PR export them directly from the `web3` package, but this seemed like a better approach to me.

Reply:
usually we move types in web3-types package if these are used by 1+ package / are common types. 
As web3-types package is a dependency of almost all other packages so we try to keep it light for usecase of other individual packages usage.

( same for utils package , for future changes)

Reply:
I believe since we created web3-types, it should encompass all the neccesary types to use the package. I think as well remain consistent, as we have EIP1193Provider types EIP6693 provider types should be included. This will allow these types to be exported through the main package which would make alot of since for most users, as I believe not very many users would know that EIP6693 types would be exported through providers namespace. We had a discussion on this a while back on this issue about doing this change https://github.com/web3/web3.js/issues/6952

Reply:
> as I believe not very many users would know that EIP6693 types would be exported through providers namespace.

I agree on this, that these types should be directly exported from web3 package, instead of via namspace. 

But regarding first point : only shared types shouldbe in web3-types package, and eip6963 types are used in web3 main package only? Same applies to other packages non-shared types like https://github.com/web3/web3.js/blob/4.x/packages/web3-eth-contract/src/types.ts

@luu-alex

Reply:
I see what you are getting at, I think it should be fine as long as this is documented and users are able to bring these types in  properly